### PR TITLE
Regenerate all APIs, and automate regeneration

### PR DESCRIPTION
**Pull request by jskeet**
This is a massive set of commits, but it all really needs to go in together.
The project changes are at the end, so I can amend that final commit to a new GAX beta when we're happy.

### CODE REVIEW

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/project.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
   "dependencies": {
     "Google.Api.CommonProtos": "1.0.0-beta02",
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/project.json]: There's some inconsistency with the Google.Api.CommonProtos version (beta02 vs beta04). Is this deliberate or should they all be the same?

File: apis/Google.Datastore.V1/Google.Datastore.V1/project.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
   "dependencies": {
     "Google.Api.CommonProtos": "1.0.0-beta02",
```
Comment (chrisdunelm) [inline apis/Google.Datastore.V1/Google.Datastore.V1/project.json]: beta04?

File: apis/Google.Devtools.Clouddebugger.V2/Google.Devtools.Clouddebugger.V2/project.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+
+  "buildOptions": {
+    "keyFile": "../../GoogleApis.snk",
+    "xmlDoc": true
+  },
+
+  "dependencies": {
+    "Google.Devtools.Source.V1": { "target": "project" },
+    "Google.Api.CommonProtos": "1.0.0-beta02",
```
Comment (chrisdunelm) [inline apis/Google.Devtools.Clouddebugger.V2/Google.Devtools.Clouddebugger.V2/project.json]: beta04?

File: apis/Google.Devtools.Clouddebugger.V2/Google.Devtools.Source.V1/project.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "requireLicenseAcceptance": false
+  },
+
+  "buildOptions": {
+    "keyFile": "../../GoogleApis.snk",
+    "xmlDoc": true
+  },
+
+  "dependencies": {
+    "Google.Api.CommonProtos": "1.0.0-beta02",
```
Comment (chrisdunelm) [inline apis/Google.Devtools.Clouddebugger.V2/Google.Devtools.Source.V1/project.json]: 04?

File: apis/Google.Devtools.Cloudtrace.V1/Google.Devtools.Cloudtrace.V1/project.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
 
   "dependencies": {
     "Google.Api.CommonProtos": "1.0.0-beta02",
```
Comment (chrisdunelm) [inline apis/Google.Devtools.Cloudtrace.V1/Google.Devtools.Cloudtrace.V1/project.json]: 04?

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/project.json (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/project.json]: It won't matter - but at some point before GA we should canonicalize, of course.

**Pull request merged**